plugins/out_kafka/kafka.c

```diff
@@ -148,16 +157,46 @@ int produce_message(struct flb_time *tm, msgpack_object *map,
     }
     else {
         size = map->via.map.size;
+        /* Make room for the message_hash */
+        if (ctx->hash) {
+            size++;
+        }
         msgpack_pack_map(&mp_pck, size);
     }
 
+    if (ctx->hash) {
+        /* Init message_hash with timestamp */
+        message_hash = (unsigned long) flb_time_to_double(tm);
+    }
     for (i = 0; i < map->via.map.size; i++) {
         key = map->via.map.ptr[i].key;
         val = map->via.map.ptr[i].val;
 
         msgpack_pack_object(&mp_pck, key);
         msgpack_pack_object(&mp_pck, val);
 
+        if (ctx->hash) {
+            if (key.via.str.size) {
+                for (counter = 0; counter < key.via.str.size; counter++) {
+                      message_hash = key.via.str.ptr[counter] + (message_hash << 5) + message_hash;
+                }
+            }
+            if (val.type == MSGPACK_OBJECT_STR) {
+                for (counter = 0; counter < val.via.str.size; counter++) {
+                      message_hash = val.via.str.ptr[counter] + (message_hash << 5) + message_hash;
+                }
+            }
+            if (val.type == MSGPACK_OBJECT_FLOAT64) {
+                message_hash = (unsigned long) (message_hash + (val.via.f64 * 1000));
+            }
+            if (val.type == MSGPACK_OBJECT_POSITIVE_INTEGER) {
+                message_hash = (unsigned long) (message_hash + (val.via.u64 << 6));
+            }
+            if (val.type == MSGPACK_OBJECT_NEGATIVE_INTEGER) {
+                message_hash = (unsigned long) (message_hash + (val.via.i64 << 6));
+            }
+        }
+
```

Comment:
I don't know how this hash algorithm is effective.
How about using `mbedtls_sha256_ret` ?